ale/isd_generate.py

```diff
@@ -51,6 +51,34 @@ def main():
              "and the default strategy of replacing their final suffix with "
              ".json will be used to generate the output file paths."
     )
+    parser.add_argument(
+        "--semimajor", "-a", "-r", "--radius",
+        required="--semiminor" in sys.argv,
```

Comment:
I tested this locally with -b and no -a. It doesn't throw an error if -b is passed without -a, but it silently uses the default radii.

The "required" option does seem to work if we pass the full "--semiminor" flag without "--semimajor" i.e. it throws an error and an explanation that both values are required.

It looks like changing this line to
```         required="--semiminor" in sys.argv or "-b" in sys.argv,```
will get this working correctly.

Reply:
ahh perfect. I fought with that for a longtime and had issues getting that to work. Good find and fix.